### ISSUE TITLE
[BUG]: Changing note type while creating a new card deletes the current content of the fields

### ISSUE THREAD

**Issue by natema** [ankidroid/Anki-Android#17476]
### Checked for duplicates?

- [x] This issue is not a duplicate

### Does it also happen in the desktop version?

- [x] This bug does not occur in the latest version of Anki Desktop

### What are the steps to reproduce this bug?

- Create a new card
- Write something in the card field (e.g. `Front` in the `Basic` type)
- Change the note type (e.g. `Basic` -> `Cloze`)

### Expected behaviour

The current content of the fields is kept as the input of the fields of the new type, as in the Desktop version. 
For example, if the current situation is
```
Type: Basic
Front: Test Front
Back: Test Back
```
and the type is changed to `Cloze` in the **Desktop version**, then the window shows 
```
Type: Cloze
Text: Test Front
Back Extra: Test Back
```
while if the type is changed to `Cloze` in the **Android version**, then the window shows 
```
Type: Cloze
Text: 
Back Extra: 
```

### Debug info

```text
AnkiDroid Version = 2.19.2 (b073f52783583fd4fd1436d986090ced9006a66b)

Backend Version = 0.1.43-anki24.06.3 (24.06.3 d678e39350a2d243242a69f4e22f5192b04398f2)

Android Version = 13 (SDK 33)

ProductFlavor = play

Manufacturer = samsung

Model = SM-T976B

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 13; SM-T976B Build/TP1A.220624.014; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/131.0.6778.39 Safari/537.36

ACRA UUID = d3f4901a-c6f6-453e-9e92-03961a0d8f39

FSRS = 0.6.4 (Enabled: false)

Crash Reports Enabled = true
```

### Research

- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by MritunjayTiwari14** [comment]
Can i work on this Issue?

**Comment by KaranJoshi1208** [comment]
Hello @natema , can I work on this issue ?

**Comment by natema** [comment]
I'm just a user, I don't have any authority in this project.

**Comment by david-allison** [comment]
@natema Was this an issue in previous versions of AnkiDroid?

**Comment by natema** [comment]
> [@natema](https://github.com/natema) Was this an issue in previous versions of AnkiDroid?

@david-allison I have a vague memory of seeing it in the past. I should be able to test some old versions next week to confirm.

**Comment by david-allison** [comment]
Old parallel versions can be found here BTW: https://github.com/ankidroid/Anki-Android/releases/tag/v2.18.4

**Comment by david-allison** [comment]
@natema Can't reproduce this. Could you provide a recording?

**Comment by natema** [comment]
@david-allison sure, here is a recording: 
https://github.com/user-attachments/assets/bbc90b85-80ce-4e3c-9e4a-9b62d5078795
Here is a screenshot of the version:
![Image](https://github.com/user-attachments/assets/5aae61f8-f504-4ab0-9dfc-edfcc133f5bd)

**Comment by natema** [comment]
@david-allison unfortunately I prefer not to share my collection. I can create a new one and do the same test while trying to add a card to that, if that helps.

**Comment by david-allison** [comment]
Totally understood, it would be fantastic to get a reproduction case

----

If not, it's been a long time since I've done it, but it shouldn't be much work to export just the collection config (deck names and settings, note types, presets etc...) without the cards

**Comment by natema** [comment]
I will try to isolate a reproducible case. In the meantime, I also tested it on iOS and I can confirm that I could **not** reproduce it.

**Comment by natema** [comment]
Just a small update that if I create a new user on the same device, install Anki and try to reproduce the bug, it doesn't occur. On the other hand, if I create a new deck on an Anki installation which exhibits the bug, it occurs also if I try to add the card to that empty deck. I vaguely remember modifying the css of the card types in the past, perhaps the issue is there. I will have a look in the next days.

**Comment by natema** [comment]
TL;DR: The issue disappeared after restoring the `Basic` and `Cloze` types to default in the `Manage Note Types` menu of the Desktop version. 

I cannot pinpoint the cause of the problem, but I resolved it. 
I had previously played with the css style on the Desktop Anki (`Tools > Manage Note Types`, select `Basic` or `Cloze` as types and then clicking on `Cards`). Trying to remove all modification (copy-pasting the same style and templates from an empty profile), and syncing the Desktop and the Android version didn't change anything. However, after restoring the `Basic` and `Cloze` types to default (for `Basic` and `Cloze` , `Tools > Manage Note Types > Cards` then `Options` on the top right and finally `Restore to Default...`) and then syncing resolved the issue.
